blockstore/splitstore/splitstore.go

```diff
@@ -1259,6 +1318,7 @@ func (s *SplitStore) walkObject(c cid.Cid, walked *cid.Set, f func(cid.Cid) erro
 	return nil
 }
 
+// Maybe check closing in here? This could run for a while.
```

Comment:
sure, that's probably a good idea.

Reply:
done, also in `walkObject` -- i put it right before scanning links for recursion.